tests/api/event-services/work-request.test.js

```diff
@@ -1,73 +1,27 @@
+const MockSocket = require('socket.io-mock');
 const handleWorkRequest = require('../../../src/api/event-services/work-request');
 const handlePagination = require('../../../src/utils/handlePagination');
-const cache = require('../../../src/cache');
-
-let mockCacheKey;
+const CacheSingleton = require('../../../src/cache');
 
 jest.mock('../../../src/utils/handlePagination');
-jest.mock('../../../src/cache', () => ({
-  get: jest.fn((key) => {
-    if (key === mockCacheKey) {
-      return {
-        results: [
-          {
-            body: JSON.stringify({
-              rows:
-                [
-                  {
-                    name: 'z',
-                  },
-                  {
-                    name: 'c',
-                  },
-                  {
-                    name: 'a',
-                  },
-                ],
-            }),
-          },
-        ],
-      };
-    }
-    return null;
-  }),
-  set: jest.fn(),
-}));
-
-let io;
-let emitSpy;
-let toSpy;
-
-const setIoMock = () => {
-  emitSpy = jest.fn();
-
-  const mockEmit = {
-    emit: emitSpy,
-  };
-  toSpy = jest.fn(() => mockEmit);
+jest.mock('../../../src/cache');
 
-  io = {
-    to: toSpy,
-    emit: emitSpy,
-  };
-};
```

Comment:
This mock is replaced by `socket.io-mock`, which provides a proper interface for these.